City-of-Helsinki/drupal-helfi-platform-config
# UHF-5610: Chat Leijuke

# Chat Leijuke [UHF-5610](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5610)
A unified chat launcher button with cookie blocked script loading.
Works with the eu_cookie_compliance module to check cookies and allow functional chat cookies implicitly on user clicking on the Leijuke.
Loads the previously implemented chat scripts and replaces their launcher buttons.
Works with multiple (tested with 2) chats/chatbots on the same page such as smartti + genesys on KYMP pysäköinti.

## What was done
* A new Chat Leijuke custom block with chat selection and optional title, that loads the leijuke library and sets js vars with the library for the selected chat.
* A new js class that is able to load a library, facilitates interaction with a cookie manager and an adapter to interact with 3rd party chat widgets. Has own cookies per chat to remember chat state open/closed.
* A drupal behaviour to instantiate the Leijuke class with eu_cookie_compliance compliant cookie manager class and a chat adapter based on the selection.
* Adapter classes for the existing chat widgets.
* Eu cookie compliance settings updated via hook.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-6277_chat_leijuke_poc`
* Run `make drush-updb drush-cr`

## How to test
* Create a new block instance from Chat Leijuke and select a chat script to load. Optionally add a title for the leijuke (translatable).
* Check the eu_cookie_compliance settings. New category 'chat' is setup and the cookie settings include: 
- * 'chat:_genesys.widgets.*'
- * 'chat:leijuke.*'
- Load page with block present and observe behavior of Leijuke and cookies.

## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/396
